app/src/main/java/com/odysee/app/OdyseeApp.java

```diff
@@ -44,4 +50,20 @@ public String getDarkModeAppSetting() {
             }
         }
     }
+
+    public ExecutorService getExecutor() {
+        if (executor == null) {
+            executor = Executors.newFixedThreadPool(4);
```

Comment:
Best practice, like on any other platform, is to use the number of cores as a maximum. My new commit checks for the amount of available cores and uses it. It will use a minimum of 4 threads.

Reply:
Good to merge now, @akinwale ?

Reply:
Yes, please.